r2i/iengine.h

```diff
@@ -58,10 +59,22 @@ class IEngine {
    * \return IPrediction inference data obtained from evaluating an IFrame
    *  on the assigned IModel.
    */
+  //[[deprecated("Use Predict(in_frame, predictions) instead.")]]
   virtual std::shared_ptr<r2i::IPrediction> Predict (std::shared_ptr<r2i::IFrame>
       in_frame,
       r2i::RuntimeError &error) = 0;
 
+  /**
+   * \brief Performs a prediction on a IEngine framework, based on a
+   *  assigned IModel.
+   *  \param in_frame Input data to perform the inference.
+   *  \param predictions [in/out] obteined from evaluating and IFrame
+   *   on the assigned IModel.
+   *  \return RuntimeError with a description of an error.
+   */
+  virtual RuntimeError Predict (std::shared_ptr<r2i::IFrame> in_frame,
```

Comment:
This change in the interface may have break test for other backends since this methods have to be implemented in the mocks. Check for example the mock of engine used in Parameters tests for the ONNXRT backend.

Reply:
You are right. Let mi check other backends tests.